File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumberServiceImpl.java
package com.example.webmvc.favouriteNumber;

import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class FavouriteNumberServiceImpl implements FavouriteNumberService {
    private final FavouriteNumberRepository favouriteNumberRepository;

    public FavouriteNumberServiceImpl(FavouriteNumberRepository favouriteNumberRepository) {
        this.favouriteNumberRepository = favouriteNumberRepository;
    }

    @Override
    public FavouriteNumber saveNumber(FavouriteNumber number) {
        return favouriteNumberRepository.saveAndFlush(number);
    }

    @Override
    public List<FavouriteNumber> getPaginated(int results_per_page, int page) {
        return favouriteNumberRepository.findAll(PageRequest.of(page, results_per_page, Sort.by("number")))
                .getContent();
    }

    @Override
    public List<FavouriteNumber> getAll() {
        return favouriteNumberRepository.findAll();
    }

    @Override
    public Optional<FavouriteNumber> findByID(long id) {
        return favouriteNumberRepository.findById(id);
    }

    @Override
    public long getNumberCount() {
        return favouriteNumberRepository.count();
    }

    @Override
    public void delete(FavouriteNumber number) {
        favouriteNumberRepository.delete(number);
    }

    @Override
    public FavouriteNumber getById(long id) {
        return favouriteNumberRepository.getById(id);
    }
}

File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumberServiceImpl.kt
package com.example.webmvc.favouriteNumber

import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort
import org.springframework.stereotype.Service
import java.util.*

@Service
class FavouriteNumberServiceImpl(
    private val favouriteNumberRepository: FavouriteNumberRepository) :
    FavouriteNumberService {
    override fun saveNumber(number: FavouriteNumber): FavouriteNumber? {
        return favouriteNumberRepository.saveAndFlush(number)
    }

    override fun getPaginated(results_per_page: Int, page: Int): List<FavouriteNumber>? {
        return favouriteNumberRepository.findAll(PageRequest.of(page, results_per_page, Sort.by("number")))
            .content
    }

    override val all: List<FavouriteNumber>
        get() = favouriteNumberRepository.findAll()

    override fun findByID(id: Long): Optional<FavouriteNumber>? {
        return favouriteNumberRepository.findById(id)
    }

    override val numberCount: Long
        get() = favouriteNumberRepository.count()

    override fun delete(number: FavouriteNumber?) {
        if (number == null) return
        favouriteNumberRepository.delete(number)
    }

    override fun getById(id: Long): FavouriteNumber? {
        return favouriteNumberRepository.getById(id)
    }
}